lib/Doctrine/DBAL/Driver/PDOPgSql/Driver.php

```diff
@@ -53,6 +53,14 @@ public function connect(array $params, $username = null, $password = null, array
                 $pdo->setAttribute(PDO::PGSQL_ATTR_DISABLE_PREPARES, true);
             }
 
+            /* defining client_encoding via SET NAMES to avoid inconsistent DSN support
+             * - the 'client_encoding' connection param only works with postgres >= 9.1
+             * - passing client_encoding via the 'options' param breaks pgbouncer support
+             */
+            if (isset($params['charset'])) {
+              $pdo->query('SET NAMES \''.$params['charset'].'\'');
```

Comment:
@deeky666 , PGSQL_LIBPQ_VERSION is showing 8.4.20 -- and client_encoding is working connecting to 9.3. So obviously checking if PGSQL_LIBPQ_VERSION is >= 9.1 in order to decide to use client_encoding (or SET NAMES) is not a reliable solution (or we'll be making the query when we don't even need to!).

I found an article earlier with somebody complaining that PGSQL_LIBPQ_VERSION wasn't properly refecting the accurate version -- and the comments mentioned that PGSQL_LIBPQ_VERSION won't reflect upgrades that are made later.

So in my case, I think PGSQL_LIBPQ_VERSION is showing 8.4 (the out of the box version) -- even though I've replaced it with 9.x.

Moral of the story: PGSQL_LIBPQ_VERSION is not a reliable constant to use for this purpose.